vcd/resource_vcd_vm_internal_disk_test.go

```diff
@@ -0,0 +1,468 @@
+  value = vcd_vapp_vm.{{.VmName}}.internal_disk[0].bus_number
+  depends_on = [vcd_vapp_vm.{{.VmName}}]
+}
+
+output "internal_disk_unit_number" {
+  value = vcd_vapp_vm.{{.VmName}}.internal_disk[0].unit_number
+  depends_on = [vcd_vapp_vm.{{.VmName}}]
+}
+
+output "internal_disk_storage_profile" {
+  value = vcd_vapp_vm.{{.VmName}}.internal_disk[0].storage_profile
+  depends_on = [vcd_vapp_vm.{{.VmName}}]
+}
+
+`
+
+const sourceTestVmInternalDiskIde = sourceTestVmInternalDiskOrgVdcAndVM + `
+resource "vcd_vm_internal_disk" "{{.DiskResourceName}}_ide" {
+  org             = "{{.Org}}"
+  vdc             =  vcd_org_vdc.{{.VdcName}}.name
+  vapp_name       = vcd_vapp.{{.VappName}}.name
+  vm_name         = vcd_vapp_vm.{{.VmName}}.name
+  bus_type        = "ide"
+  size_in_mb      = "{{.Size}}"
+  bus_number      = "0"
+  unit_number     = "0"
+  storage_profile = "{{.StorageProfileName}}"
+  allow_vm_reboot = "false"
+}
+`
+
+const sourceTestVmInternalDisk = sourceTestVmInternalDiskOrgVdcAndVM + `
+resource "vcd_vm_internal_disk" "{{.DiskResourceName}}_ide" {
+  org             = "{{.Org}}"
+  vdc             =  vcd_org_vdc.{{.VdcName}}.name
+  vapp_name       = vcd_vapp.{{.VappName}}.name
+  vm_name         = vcd_vapp_vm.{{.VmName}}.name
+  bus_type        = "ide"
+  size_in_mb      = "{{.Size}}"
+  bus_number      = "0"
+  unit_number     = "0"
+  storage_profile = "{{.StorageProfileName}}"
+  allow_vm_reboot = "true" 
+}
+
+resource "vcd_vm_internal_disk" "{{.DiskResourceName}}" {
+  org             = "{{.Org}}"
+  vdc             =  vcd_org_vdc.{{.VdcName}}.name
+  vapp_name       = vcd_vapp.{{.VappName}}.name
+  vm_name         = vcd_vapp_vm.{{.VmName}}.name
+  bus_type        = "{{.BusType}}"
+  size_in_mb      = "{{.Size}}"
+  bus_number      = "{{.BusNumber}}"
+  unit_number     = "{{.UnitNumber}}"
+  storage_profile = "{{.StorageProfileName}}"
+  allow_vm_reboot = "false"
+}
+`
+
+const sourceTestVmInternalDisk_Update1 = sourceTestVmInternalDiskOrgVdcAndVM + `
+resource "vcd_vm_internal_disk" "{{.DiskResourceName}}" {
+  org             = "{{.Org}}"
+  vdc             =  vcd_org_vdc.{{.VdcName}}.name
+  vapp_name       = vcd_vapp.{{.VappName}}.name
+  vm_name         = vcd_vapp_vm.{{.VmName}}.name
+  bus_type        = "{{.BusType}}"
+  size_in_mb      = "{{.SizeBigger}}"
+  bus_number      = "{{.BusNumber}}"
+  unit_number     = "{{.UnitNumber}}"
+  storage_profile = "{{.StorageProfileName}}"
+  allow_vm_reboot = "false"
+}
+
+resource "vcd_vm_internal_disk" "{{.DiskResourceName}}_ide" {
+  org             = "{{.Org}}"
+  vdc             =  vcd_org_vdc.{{.VdcName}}.name
+  vapp_name       = vcd_vapp.{{.VappName}}.name
+  vm_name         = vcd_vapp_vm.{{.VmName}}.name
+  bus_type        = "ide"
+  size_in_mb      = "{{.SizeBigger}}"
+  bus_number      = "0"
+  unit_number     = "0"
+  storage_profile = "{{.StorageProfileName}}"
+  allow_vm_reboot = "true"
+}
+`
+
+/*const sourceTestVmInternalDisk = sataDisk + `
```

Comment:
Is this still needed?

Reply:
will remove when I sure we finished changing design